src/routes/safe/components/Apps/components/AddAppForm/index.tsx

```diff
@@ -95,9 +89,12 @@ const AddApp = ({ appList, closeModal }: AddAppProps): ReactElement => {
       ...persistedAppList.map(({ url, disabled }) => ({ url, disabled })),
     ]
     saveToStorage(APPS_STORAGE_KEY, newAppList)
-    const goToApp = `${appsPath}?appUrl=${encodeURI(appInfo.url)}`
-    history.push(goToApp)
-  }, [appInfo.url, history, appsPath])
+
+    history.push({
+      pathname: generateSafeRoute(SAFE_ROUTES.APPS, getPrefixedSafeAddressFromUrl()),
+      search: `?appUrl=${encodeURI(appInfo.url)}`,
```

Comment:
Btw this should probably be `encodeURIComponent`, otherwise apps with a search param in the URL will break our app.
Just a side note, no need to change it here.